src/Hedwig/Repositories/HedwigRepository.cs

```
@@ -214,11 +214,23 @@ private bool IdEquals(object a, object b)
 			var aIdProp = a.GetType().GetProperty("Id");
 			var bIdProp = b.GetType().GetProperty("Id");
 
-			if (aIdProp == null || bIdProp == null || aIdProp.PropertyType != bIdProp.PropertyType)
+			// Guard for types without Ids or mismatched Id types
+			// (Should never happen)
+			if (aIdProp == null
+				|| bIdProp == null
+				|| aIdProp.PropertyType != bIdProp.PropertyType
+			)
 			{
 				return false;
```

Do we want to throw an exception here instead if this should never happen?